pkg/api/util/names.go

```diff
@@ -22,30 +22,33 @@ import (
 	"hash/fnv"
 
 	"regexp"
-
-	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
 )
 
-func ComputeCertificateRequestName(crt *cmapi.Certificate) (string, error) {
-	specBytes, err := json.Marshal(crt.Spec)
+func ComputeName(prefix string, obj interface{}) (string, error) {
```

Comment:
Add a comment to this function explaining the name requirements and linking to an explanation of the name length limits.

My concern is that the DNSSafeShortenTo52Characters stuff may not be necessary and I want to understand the motivation for it.
It seems to me the the name limit is actually 253 chars, as described here: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

The DNS label length limit only applies to resources where the name MUST be a valid DNS label, rather than a DNS domain name.

Reply:
These resources are used as the prefix for creating Service, Ingress and Pod resources in the case of ACME. This is where the 63 char limit was 'inherited' from.

Reply:
In that case, perhaps we should apply the truncation when we create those service names and ingress names rather than when creating the CertificateRequest, Order names.

Happy to add a `lgtm` to this now that I understand the reasons, but @meyskens I think we should add a comment or two, recording what @munnerz explained above.